sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_services_first_seen_v1/query.sql

```diff
@@ -1,136 +1,99 @@
-WITH base AS (
+WITH fxa_events AS (
   SELECT
-    * REPLACE (
-      IF(service IS NULL AND event_type = 'fxa_activity - cert_signed', 'sync', service) AS service
-    )
+    timestamp,
+    user_id,
+    IF(service IS NULL AND event_type = 'fxa_activity - cert_signed', 'sync', service) AS service,
+    device_id,
+    os_name,
+    flow_id,
+    event_type,
+    country,
+    LANGUAGE,
+    entrypoint,
+    utm_term,
+    utm_medium,
+    utm_source,
+    utm_campaign,
+    utm_content,
+    ua_version,
+    ua_browser,
   FROM
-    firefox_accounts.fxa_content_auth_oauth_events
+    `moz-fx-data-shared-prod.firefox_accounts.fxa_content_auth_oauth_events`  -- # TODO: there seems to be fxa_content_auth_events, maybe we should use that instead since oauth has been deprecated for a long time now?
+  WHERE
+    DATE(timestamp)
+    BETWEEN DATE_SUB(@submission_date, INTERVAL 1 DAY)
+    AND @submission_date
 ),
-  -- use a window function to look within each USER and SERVICE for the first value of service, os, and country.
-  -- also, get the first value of flow_id for later use and create a boolean column that is true if the first instance of a service usage includes a registration.
-  -- [kimmy] the variable first_service_timestamp_last is named so because it is actually the last timestamp recorder in the user's first flow,
-  -- NOT the first timestamp in their first flow.
-  -- it's used later on to order by service, so i'm keeping it here and just renaming it.
-first_services AS (
-  SELECT
-    ROW_NUMBER() OVER w1_unframed AS _n,
+new_user_service_os_entries AS (
+  SELECT DISTINCT
+    flow_id,
+    timestamp,
     user_id,
     service,
-      -- using mode_last with w1_reversed to get mode_first
-    udf.mode_last(ARRAY_AGG(`timestamp`) OVER w1_reversed) AS first_service_timestamp_last,
-    udf.mode_last(ARRAY_AGG(flow_id) OVER w1_reversed) AS first_service_flow,
-    LOGICAL_OR(IFNULL(event_type = 'fxa_reg - complete', FALSE)) OVER w1_reversed AS did_register
+    device_id,
+    os_name,
+    country,
+    LANGUAGE,
+    entrypoints.entrypoint,
+    utm_term,
+    utm_medium,
+    utm_source,
+    utm_campaign,
+    utm_content,
+    ua_version,
+    ua_browser,
   FROM
-    base
+    fxa_events
+  LEFT JOIN
+    (SELECT DISTINCT flow_id, entrypoint FROM fxa_events) AS entrypoints
+  USING
+    (flow_id)
   WHERE
-    (
-      (event_type IN ('fxa_login - complete', 'fxa_reg - complete') AND service IS NOT NULL)
-      OR (event_type LIKE 'fxa_activity%')
+    DATE(timestamp) = @submission_date
+    AND flow_id IS NOT NULL
+    AND ((event_type IN ('fxa_login - complete', 'fxa_reg - complete') AND service IS NOT NULL))
+    AND CONCAT(user_id, service, device_id) NOT IN (
```

Comment:
IIUC this is to ensure we are not adding additional entries for a previously-seen user-service-device combination. this means that if there ARE previous entries for **a previously seen service, on a different DEVICE** then we WILL insert another row. 

i can see the utility of this from the sync perspective, since it (kind of) tells us when a new device was added to sync (it is not 100% accurate since IIUC in some cases a device that is disconnected from sync but then reconnected will get a fresh device_id - i am not sure of the details here).

assuming we want to go this route, should we rename the table `fxa_users_services_devices_first_seen_v1` ?

Reply:
i agree, we should rename if using device_id like this

Reply:
100% agree with this, already mentioned this in one of my previous comments. The dataset has been renamed to `fxa_users_services_devices_first_seen_v1` as suggested by @irrationalagent

Reply:
ok so naming the query results in the following task name `firefox_accounts_derived__fxa_users_services_devices_first_seen__v1` (67 chars) which exceeds the char limit of `62` and makes the DAG generation fail...

@relud I wonder if we should make a change to bqetl dag generation to truncate the task name to `62` characters during DAG generation but allow actual dataset names to be longer?

Reply:
@relud would something like this be appropriate?
https://github.com/mozilla/bigquery-etl/pull/3279

Reply:
yes, see review there for more details

Reply:
thanks @relud 🙇 The change in bqetl has been applied.